Source/Netvars.cpp

```diff
@@ -24,13 +24,18 @@
 
 static std::unordered_map<std::uint32_t, std::pair<recvProxy, recvProxy*>> proxies;
 
-static void CDECL_CONV spottedHook(recvProxyData& data, void* arg2, void* arg3) noexcept
+static void CDECL_CONV spottedHook(recvProxyData& data, void* outStruct, void* arg3) noexcept
 {
-    if (Misc::isRadarHackOn())
+    const auto entity = reinterpret_cast<Entity*>(outStruct);
+
+    if (Misc::isRadarHackOn()) {
         data.value._int = 1;
 
+        entity->spottedByMask() |= 1 << (localPlayer->index() - 1);
```

Comment:
1. `localPlayer` should be checked for null pointer
2. `localPlayer->index()` should be <= 32 to not overflow when bit shifting